source/backtraceMetric.ts

```diff
@@ -0,0 +1,226 @@
+import { BacktraceClientOptions } from '.';
+import { uuid } from './utils';
+
+declare const __VERSION__: string;
+
+/**
+ * Function to create instance of BacktraceMetric and ping.
+ */
+export default function backtraceMetric(
+  configuration: BacktraceClientOptions,
+): void {
+  new BacktraceMetric(configuration).ping();
+}
+
+/**
+ * Handles Backtrace Metrics.
+ * TODO: Consider converting to a static class since an instance is only created to call a single function once.
+ */
+export class BacktraceMetric {
+  public universe;
+  public token;
+  public timeout = 15000; // fifteen seconds in miliseconds
+
+  /** Seconds since epoch. */
+  public readonly CURRENT_TIMESTAMP: number = Math.floor(
+    new Date().getTime() / 1000,
+  );
+  /** Thirty minutes in seconds. */
+  public readonly THIRTY_MINUTES = 1800;
+
+  public readonly userAgent = navigator.userAgent;
+  public readonly applicationName = 'backtrace-js';
+  public readonly applicationVersion = __VERSION__;
+
+  // These fields are stored in localStorage for persistence.
+  public sessionId = this.getSessionId();
+  public lastActive = this.getLastActive();
```

Comment:
I would suggest to write a getter instead - right now users can override this value. Is there any reason why it shouldn't be getter/readonly for users

Reply:
I removed these class variables since they didn't provide any functionality, and stored potentially outdated values.
These are "private" values so no need for a real `get` (for now).